### ISSUE TITLE
Lint errrors in .pub-cache

### ISSUE THREAD

**Issue by DGoiana** [invertase/dart_custom_lint#280]
**Description**

Running the linter, some problems appear in the `.pub-cache` folder.

**To Reproduce**

Run `dart run custom_lint`

```
Error from the analysis server: /home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/add_cubit_suffix_for_cubits.dart:4:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/avoid_conditional_hooks.dart:5:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/avoid_single_child_in_multi_child_widget.dart:4:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/catch_parameter_names.dart:4:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/constructor_parameters_and_fields_should_have_the_same_order.dart:4:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/hook_widget_does_not_use_hooks.dart:5:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/prefix_widgets_returning_slivers.dart:4:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/start_comments_with_space.dart:4:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/use_instead_type.dart:5:1: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
import 'package:custom_lint_builder/custom_lint_builder.dart';
^^^^^^^^
/home/runner/.pub-cache/hosted/pub.dev/leancode_lint-13.0.0/lib/lints/add_cubit_suffix_for_cubits.dart:57:12: Error: 'LintCode' is imported from both 'package:analyzer/src/dart/error/lint_codes.dart' and 'package:custom_lint_core/src/lint_codes.dart'.
    return LintCode(
           ^^^^^^^^


Error from the analysis server: Failed to start plugins
```

**Expected behavior**

Is it supposed to run in cached files?

**Comment by rrousselGit** [comment]
Eh, how did you even get into this state?

Could you share a git repository to reproduce?

**Comment by DGoiana** [comment]
Sure, https://github.com/NIAEFEUP/uni/actions/runs/11240916932/job/31251329009?pr=1348

**Comment by FaFre** [comment]
I can add this ones upon running `build_runner` after updating to latest master (& upgrading deps):

```bash
[WARNING] ../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:92:41: Error: Type 'AugmentationImportDirective' not found.
  void visitAugmentationImportDirective(AugmentationImportDirective node) {
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:686:42: Error: Type 'LibraryAugmentationDirective' not found.
  void visitLibraryAugmentationDirective(LibraryAugmentationDirective node) {
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:1354:28: Error: Type 'AugmentationImportDirective' not found.
  final List<_Subscription<AugmentationImportDirective>>
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:2025:28: Error: Type 'LibraryAugmentationDirective' not found.
  final List<_Subscription<LibraryAugmentationDirective>>
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:1357:33: Error: Type 'AugmentationImportDirective' not found.
      String key, void Function(AugmentationImportDirective node) listener) {
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:2028:33: Error: Type 'LibraryAugmentationDirective' not found.
      String key, void Function(LibraryAugmentationDirective node) listener) {
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:2736:21: Error: Type 'AugmentationImportDirective' not found.
      void Function(AugmentationImportDirective node) listener) {
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:3270:21: Error: Type 'LibraryAugmentationDirective' not found.
      void Function(LibraryAugmentationDirective node) listener) {
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:92:41: Error: 'AugmentationImportDirective' isn't a type.
  void visitAugmentationImportDirective(AugmentationImportDirective node) {
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:94:11: Error: Superclass has no method named 'visitAugmentationImportDirective'.
    super.visitAugmentationImportDirective(node);
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:686:42: Error: 'LibraryAugmentationDirective' isn't a type.
  void visitLibraryAugmentationDirective(LibraryAugmentationDirective node) {
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:688:11: Error: Superclass has no method named 'visitLibraryAugmentationDirective'.
    super.visitLibraryAugmentationDirective(node);
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:1354:28: Error: 'AugmentationImportDirective' isn't a type.
  final List<_Subscription<AugmentationImportDirective>>
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:1357:33: Error: 'AugmentationImportDirective' isn't a type.
      String key, void Function(AugmentationImportDirective node) listener) {
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:2025:28: Error: 'LibraryAugmentationDirective' isn't a type.
  final List<_Subscription<LibraryAugmentationDirective>>
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:2028:33: Error: 'LibraryAugmentationDirective' isn't a type.
      String key, void Function(LibraryAugmentationDirective node) listener) {
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:2736:21: Error: 'AugmentationImportDirective' isn't a type.
      void Function(AugmentationImportDirective node) listener) {
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/custom_lint_core-0.6.5/lib/src/node_lint_visitor.g.dart:3270:21: Error: 'LibraryAugmentationDirective' isn't a type.
      void Function(LibraryAugmentationDirective node) listener) {
```

Flutter 3.26.0-1.0.pre.414 • channel master • https://github.com/flutter/flutter.git
Framework • revision c077b8c61a (9 hours ago) • 2024-10-08 19:51:19 -0400
Engine • revision 0e7344ae24
Tools • Dart 3.6.0 (build 3.6.0-332.0.dev) • DevTools 2.40.0

**Comment by rrousselGit** [comment]
> https://github.com/NIAEFEUP/uni/actions/runs/11240916932/job/31251329009?pr=1348

That's unrelated. It's the job of leancode_lint to fix its import conflict.

**Comment by rrousselGit** [comment]
Fixed in 0.6.9
